src/v/pandaproxy/schema_registry/protobuf.cc

```diff
@@ -499,14 +499,33 @@ struct compatibility_checker {
             }
         }
 
+        // check writer fields
         for (int i = 0; i < writer->field_count(); ++i) {
+            // TODO(oren): Need special handling here?
             if (reader->IsReservedNumber(i) || writer->IsReservedNumber(i)) {
```

Comment:
fwiw, I don't think I ever got to understanding what this means in the context of compatibility checking. may not be so relevant here (nothing has changed) but maybe Ben can comment.

Reply:
>If the field is required by the reader, but reserved by the writer, it should be incompatible.

That sounds right...maybe the converse as well? I didn't experiment w/ this at all, so maybe a quick test against the reference impl would be prudent.

At any rate this block does need some special treatment then, right? Rather than skipping over reserved fields, it seems like reserved-ness should be part of the predicate below.

Reply:
I don't think `reserved`ness should be part of the compat checks. `reserved` is just an annotation in protobuf, it just ensures that another field cannot be defined with the same tag number / name to help ensure that old fields are not reintroduced.

> If the field is required by the reader, but reserved by the writer, it should be incompatible.

This is true because "reserved by the writer" means that the writer doesn't have the tag and so it is just like introducing a required field which is incompatible.

Reply:
oh yeah, even better. point being that we shouldn't be skipping checks on that basis. 👍